pkg/controller/controller_utils.go

```diff
@@ -430,14 +438,35 @@ func GetPodFromTemplate(template *api.PodTemplateSpec, parentObject runtime.Obje
 			GenerateName: prefix,
 		},
 	}
+	if setControllerRef {
```

Comment:
why not just make this a different method? passing a boolean flag to a public method is sometimes confusing. Soon enough that'll become a struct of bools.